bibtex-completion.el

```diff
@@ -1254,56 +1285,77 @@ line."
-  (setq-local
-   header-line-format nil)
+  (setq-local header-line-format nil)
   (save-buffer)
   (let ((window (get-buffer-window (file-name-nondirectory bibtex-completion-notes-path))))
     (if (and window (not (one-window-p window)))
         (delete-window window)
       (switch-to-buffer (other-buffer)))))
 
-(defun bibtex-completion-edit-notes (keys)
-  "Open the notes associated with the selected entries using `find-file'."
+(defun bibtex-completion-edit-notes-multiple-files (keys)
+  "Using one notes file per publication, open notes associated with the selected entries."
   (dolist (key keys)
-    (let* ((entry (bibtex-completion-get-entry key))
-           (year (or (bibtex-completion-get-value "year" entry)
-                     (car (split-string (bibtex-completion-get-value "date" entry "") "-"))))
-           (entry (push (cons "year" year) entry)))
-      (if (and bibtex-completion-notes-path
-               (f-directory? bibtex-completion-notes-path))
-                                        ; One notes file per publication:
-          (let* ((path (f-join bibtex-completion-notes-path
-                               (s-concat key bibtex-completion-notes-extension))))
-            (find-file path)
-            (unless (f-exists? path)
-              (insert (s-format bibtex-completion-notes-template-multiple-files
-                                'bibtex-completion-apa-get-value
-                                entry))))
-                                        ; One file for all notes:
+    (let ((entry (bibtex-completion-get-entry key))
+	  (path (f-join bibtex-completion-notes-path
+			(s-concat key bibtex-completion-notes-extension))))
+      ;; Why not `find-file-other-window'?
+      (find-file path)
+      (unless (f-exists? path)
+	(insert (s-format bibtex-completion-notes-template-multiple-files
+                          'bibtex-completion-apa-get-value
+                          entry))))))
+
+(defun bibtex-completion-clean-notes-string (s)
+  "Unescape ampersand and remove brackets from S."
+  (--> (replace-regexp-in-string "\\\\&" "&" s)
+       (replace-regexp-in-string "[\\\{}]+" "" it)))
+
+(defun bibtex-completion-capture-placeholder ()
+  "Placeholder for `bibtex-completion-capture-template'."
+  (bibtex-completion-clean-notes-string
+   (s-format bibtex-completion-notes-template-one-file
+	     'bibtex-completion-apa-get-value entry)))
+
+(defun bibtex-completion-edit-notes-one-file (keys)
+  "Using one file for all notes, open notes associated with the selected entries.
+If notes are not found, create one using org-capture (see
+`bibtex-completion-capture-template')."
+  (dolist (key keys)
+    (let ((entry (bibtex-completion-get-entry key)))
+      (if (not (assoc "=has-note=" entry))
+	  ;; No notes found, so create one
+          (progn
+            (require 'org-capture)
+            (unless (assoc-default "bibtex" org-capture-templates)
+              (cl-pushnew bibtex-completion-capture-template
```

Comment:
Would it make sese to remove "bibtex" from `org-capture-templates` after completion of `bibtex-completion-edit-notes-one-file`? My concern is, that running `org-capture` separately from `bibtex-completion-edit-notes-one-file` should not show the "bibtex" action. (I haven't tested if this is the case with your code, but it seems to be the case?)

Reply:
Good point. I agree.

Reply:
Even better, could we just let-bind `org-capture-templates`?

Reply:
I use let binding since opening https://github.com/tmalsburg/helm-bibtex/issues/260 and didn't observe any interference with "normal" org-capture use.